srlife/damage.py

```diff
@@ -118,6 +126,158 @@ def tube_log_reliability(self, tube, material, receiver):
         # Return the sums as a function of time
         return np.sum(inc_prob, axis=1)
 
+
+class CrackShapeIndependent(WeibullFailureModel):
+    """
+    Parent class for crack shape dependent models
+    """
+
+    def __init__(self, temperatures, material):
+        """
+        Input the material parameters
+        Create a mesh grid for integration
+        Evaluate direction cosines using the mesh grid
+        only for PIA and WNTSA models which are crack shape independent
+        """
+        super().__init__(temperatures, material)
+        self.temperatures = temperatures
+        self.material = material
+        # Material parameters
+        self.svals = self.material.strength(temperatures)
```

Comment:
You need to calculate all these values (which depend on the material and temperature) in `calculate_element_log_reliability`